cli/src/main.rs

```diff
@@ -80,7 +79,11 @@ pub async fn main() -> ExitCode {
 pub async fn run_wasm_main(opts: Opts) -> Result<(), anyhow::Error> {
     // Load the wasm module into an execution context
     let ctx = create_execution_context(&opts).await?;
-    ctx.run_main(opts.wasm_args()).await
+    ctx.run_main(
+        opts.input().file_stem().unwrap().to_str().unwrap(),
```

Comment:
I think the second `unwrap` here could cause us problems since Windows filenames are, iirc, UTF-16. Can you use `to_string_lossy` instead?

The first `unwrap` should be okay, because presumably `check_module` would fail if this argument was a directory. It might be worth saying so explicitly, though.

Reply:
@acfoltzer I think you're right that `check_module` would catch that while parsing CLI args. I added a message here just to make it a bit more clear what is going on though.

Reply:
also I'm still fairly new to Rust, so if there's a different way that you would have written this, please let me know :)

Reply:
I think this is fine, especially since it'd come up right away when executing a command rather than lying in wait to bring down a long-running system. I could imagine introducing a new type for the validated filename just to have something tracked by the compiler to this point assuring us that the filename exists, but it'd honestly be overkill for this situation.